### ISSUE TITLE
Gregorian Date Editor Type allows 0's in day and month fields 

### ISSUE THREAD

**Issue by emaayan** [davetcc/tcMenu#332]
**Describe the bug**
Gregorian Date Editor Type allows 0's in day and month fields 
for example 00/00/2020 

**Commerical or Personal use**
personal 

**To Reproduce**
include the following emf file 

**Expected behavior**
values should only chane the minimum allowed 
for example 01/01/2000 

[menuShow.emf.txt](https://github.com/davetcc/tcMenu/files/11221574/menuShow.emf.txt)

**Screenshots**
![image](https://user-images.githubusercontent.com/13434544/231744214-666a4887-4477-46d7-bf61-d860d35055ce.png)

**Embedded environment
esp32 dev board 
diymore 2.42 screen 

code being used:

```
#include "menuShow_menu.h"

#define CS 5
#define DC 17   // 16
#define RESET 4 // 17
U8G2_SSD1309_128X64_NONAME0_F_4W_HW_SPI _oled(U8G2_R0, CS, DC, RESET);


void setup() {
    _oled.begin();
    setupMenu();   
}
void loop() {
    taskManager.runLoop();    

}


```

**Comment by davetcc** [comment]
See https://github.com/davetcc/tcMenuLib/issues/194

**Comment by davetcc** [comment]
Library issue fixed for 3.2

**Comment by emaayan** [comment]
is 3.2 available?  (i saw the beta of the designer available)
